presto-native-execution/presto_cpp/main/http/tests/HttpJwtTest.cpp

```diff
@@ -0,0 +1,199 @@
+    for (const auto& [configName, configValue] : configOverride) {
+      systemConfig[configName] = configValue;
+    }
+
+    return std::make_unique<core::MemConfigMutable>(systemConfig);
+  }
+
+  void setupJwtNodeConfig() const {
+    std::unordered_map<std::string, std::string> nodeConfigValues{
+        {std::string(NodeConfig::kMutableConfig), std::string("true")},
+        {std::string(NodeConfig::kNodeId), std::string("testnode")}};
+    std::unique_ptr<Config> rawNodeConfig =
+        std::make_unique<core::MemConfig>(nodeConfigValues);
+    nodeConfig_->initialize(std::move(rawNodeConfig));
+  }
+
+  std::unique_ptr<http::HttpResponse> produceHttpResponse(
+      const bool useHttps,
+      std::unordered_map<std::string, std::string> clientSystemConfigOverride =
+          {},
+      std::unordered_map<std::string, std::string> serverSystemConfigOverride =
+          {},
+      const uint64_t sendDelayMs = 500) {
+    auto clientConfig = jwtSystemConfig(clientSystemConfigOverride);
+    auto systemConfig = SystemConfig::instance();
+    systemConfig->initialize(std::move(clientConfig));
+
+    auto server = getServer(useHttps);
+
+    auto request = std::make_shared<AsyncMsgRequestState>();
+
+    // Set the delay to 1s to have enough time to update the config.
+    server->registerGet("/async/msg", asyncMsg(request));
+
+    std::vector<std::unique_ptr<proxygen::RequestHandlerFactory>> filters;
+    filters.push_back(
+        std::make_unique<http::filters::InternalAuthenticationFilterFactory>());
+    HttpServerWrapper wrapper(std::move(server));
+    wrapper.setFilters(filters);
+    auto serverAddress = wrapper.start().get();
+
+    HttpClientFactory clientFactory;
+    // Make sure client doesn't timeout for a delayed request.
+    auto client = clientFactory.newClient(
+        serverAddress, std::chrono::milliseconds(1'000), useHttps, memoryPool_);
+
+    auto [reqPromise, reqFuture] = folly::makePromiseContract<bool>();
+    request->requestPromise = std::move(reqPromise);
+
+    auto responseFuture =
+        sendGet(client.get(), "/async/msg", sendDelayMs, "TestBody");
+
+    auto serverConfig = jwtSystemConfig(serverSystemConfigOverride);
+    auto valuesMap = serverConfig->valuesCopy();
+    /// The request is delayed. Meanwhile update the config so the server
+    /// might use a different configuration, if provided.
+    for (auto key : valuesMap) {
+      systemConfig->setValue(key.first, key.second);
+    }
+
+    /// Wait and wake up after 1s, if needed, because the server should have
+    /// rejected the request or processed the request already.
+    std::move(reqFuture).wait(std::chrono::milliseconds(1'000));
+
+    if (auto msgPromise = request->msgPromise.lock()) {
+      msgPromise->promise.setValue("Success");
+    }
+
+    auto response = std::move(responseFuture).get();
+    wrapper.stop();
+
+    return response;
+  }
+
+  const std::shared_ptr<MemoryPool> memoryPool_ =
```

Comment:
This is only used once inside `produceHttpResponse`. Let's remove this and get the defaultMemoryManager at the call site.